### DISCUSSION TITLE
HuBMAP: Registration User Interface (RUI)

### DISCUSSION THREAD

**Discussion by andreasbueckle** [nih-cfde/training-and-engagement#415] [Common Fund Tools]
In this board, please ask your questions and/or suggestions for improving the functionality of the RUI so we can then forward them to our developer team. 

Link to the RUI: https://hubmapconsortium.github.io/ccf-ui/rui/

**Comment by s-canchi** [comment]
Does the previous registered block functionality depend on the history of the browser being preserved ?

**Reply by andreasbueckle** [reply]
Yes, the RUI accesses the browser's cache to construct a visualization of previously registered blocks. @bherr2 might have to confirm.

**Comment by s-canchi** [comment]
Are there any current practical uses for the RUI that you have used ?

**Reply by emquardokus** [reply]
The RUI is used to register tissue blocks extracted from organs to mm resolution.  The tissue block will be further thin sectioned and those sections used to run various types of spatially resolved biomolecular assays (e.g. CODEX (a type of multiplexed immunofluorescence microscopy), spatial transcriptomics, or imaging mass spectrometry etc.  The spatially registered block information will provide the ability to register the subsequent biomolecular data back into that block via a second stage 2 registration currently under development. The 3D reference organ models and registered tissue blocks are placed within the context of the human body to construct an atlas.

**Comment by s-canchi** [comment]
There are “repeated” sub-structures within the kidney tissue, how does the registry represent  a tissue block  from a such repeated structure?

**Reply by emquardokus** [reply]
When a tissue block is registered, it is located in the region of the kidney where it was extracted.  In the reference 3D model of the kidney, for example,  there are many renal pyramid (a repeating structure) that exist.  The tissue block to be registered would collide with one of the repeating structures that would be within the correct region of the kidney.  The resulting JSON file provides the Cartesian coordinates for that registered block.  In short, repeated structures exist in the 3D reference organ as they would in the real organ.

**Comment by s-canchi** [comment]
Do you have a manuscript/ README on how you perform tissue/block imaging registration?

**Reply by emquardokus** [reply]
There is an embedded tutorial within the RUI https://hubmapconsortium.github.io/ccf-ui/rui/ ; click the play button to view it.

**Reply by andreasbueckle** [reply]
We also have this SOP on Using the RUI:
https://docs.google.com/document/d/11jKl__ltdDO3PBMHgHpZnIcZTNuxGUpX_94l6CtTP2I/edit?usp=sharing
